server/catgenome/src/main/java/com/epam/catgenome/util/motif/AlterDuoMotifSearchIterator.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.catgenome.util.motif;
+
+import com.epam.catgenome.entity.reference.motif.Motif;
+import com.epam.catgenome.manager.gene.parser.StrandSerializable;
+
+import java.util.Iterator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AlterDuoMotifSearchIterator implements Iterator<Motif> {
+
+    private final Matcher matcherPos;
+    private final Matcher matcherNeg;
+    private int currentStartPositionPos;
+    private int currentStartPositionNeg;
+    private final String contig;
+    private final int offset;
+    private final boolean includeSequence;
+
+    private boolean posFlag;
+    private boolean negFlag;
+
+    public AlterDuoMotifSearchIterator(final byte[] seq, final String iupacRegex, final String contig,
+                                       final int start, final boolean includeSequence) {
+
+        matcherPos = Pattern.compile(IupacRegexConverter.convertIupacToRegex(iupacRegex), Pattern.CASE_INSENSITIVE)
+                .matcher(new String(seq));
+        matcherNeg = Pattern.compile(
+                        IupacRegexConverter.convertIupacToComplementReversedRegex(iupacRegex),
+                        Pattern.CASE_INSENSITIVE)
+                .matcher(new String(seq));
```

Comment:
I believe we can introduce variable from `new String(seq)` and use it twice instead of creating String several times

Reply:
Yes, I totally agree. It will be more efficient and cut off memory usage.